infrastructure/server/oidc_idp_bcsc.tf

```diff
@@ -0,0 +1,102 @@
+  provider_name = "DEV-BCSC"
+  provider_type = "OIDC"
+
+  provider_details = {
+    authorize_scopes          = "openid profile email address"
+    client_id                 = var.dev_oidc_bcsc_idp_client_id
+    client_secret             = var.dev_oidc_bcsc_idp_client_secret
+    oidc_issuer               = var.dev_bcsc_oidc_idp_issuer
+    attributes_request_method = "GET"
+    authorize_url = "https://idtest.gov.bc.ca/login/oidc/authorize"
+    token_url = "https://idtest.gov.bc.ca/oauth2/token"
+    attributes_url = "${var.dev_bcsc_userinfo_proxy_endpoint}"
+    jwks_uri = "https://idtest.gov.bc.ca/oauth2/jwk.json"
+  }
+
+  attribute_mapping = {
+    username = "sub",
+    given_name = "given_name",
+    family_name = "family_name",
+    birthdate = "birthdate",
+    gender = "gender",
+    email                      = "email",
+    email_verified = "email_verified",
+    address = "address",
+    "custom:idp_name"          = "aud",
+    "custom:idp_user_id"       = "sub",
+    "custom:idp_display_name"  = "display_name",
+    "custom:given_names" = "given_names"
+  }
+}
+
+resource "aws_cognito_identity_provider" "test_bcsc_oidc_provider" {
+  user_pool_id  = aws_cognito_user_pool.fam_user_pool.id
+  provider_name = "TEST-BCSC"
+  provider_type = "OIDC"
+
+  provider_details = {
+    authorize_scopes          = "openid profile email address"
+    client_id                 = var.test_oidc_bcsc_idp_client_id
+    client_secret             = var.test_oidc_bcsc_idp_client_secret
+    oidc_issuer               = var.test_bcsc_oidc_idp_issuer
+    attributes_request_method = "GET"
+    authorize_url = "https://idtest.gov.bc.ca/login/oidc/authorize"
+    token_url = "https://idtest.gov.bc.ca/oauth2/token"
+    attributes_url = "${var.test_bcsc_userinfo_proxy_endpoint}"
+    jwks_uri = "https://idtest.gov.bc.ca/oauth2/jwk.json"
+  }
+
+  attribute_mapping = {
+    username = "sub",
+    given_name = "given_name",
+    family_name = "family_name",
+    birthdate = "birthdate",
+    gender = "gender",
+    email                      = "email",
+    email_verified = "email_verified",
+    address = "address",
+    "custom:idp_name"          = "aud",
+    "custom:idp_user_id"       = "sub",
+    "custom:idp_display_name"  = "display_name",
+    "custom:given_names" = "given_names"
+  }
+
+}
+
+resource "aws_cognito_identity_provider" "prod_bcsc_oidc_provider" {
+  user_pool_id  = aws_cognito_user_pool.fam_user_pool.id
+  provider_name = "PROD-BCSC"
+  provider_type = "OIDC"
+
+  provider_details = {
+    authorize_scopes          = "openid profile email address"
+    client_id                 = var.prod_oidc_bcsc_idp_client_id
+    client_secret             = var.prod_oidc_bcsc_idp_client_secret
+    oidc_issuer               = var.prod_bcsc_oidc_idp_issuer
+    attributes_request_method = "GET"
+    authorize_url = "https://idtest.gov.bc.ca/login/oidc/authorize"
+    token_url = "https://idtest.gov.bc.ca/oauth2/token"
```

Comment:
the authorize_ur, token_url and jwks_uri are all the same for dev, test and prod?